### PR TITLE
Allow for configuration without a local carrier.toml by passing config to `Conduit::new()`

### PR DESCRIPTION
**Pull request by ewilken**

### CODE REVIEW

File: src/conduit.rs [hunk 1]
```diff
@@ -51,8 +51,7 @@ impl Conduit {
     /// create a new conduit as a future.
     /// it will subscribe to the shadow address and maintain a connection to all peers on it
     #[osaka]
-    pub fn new(poll: osaka::Poll, shadow: identity::Address) -> Result<Self, Error> {
-        let config = config::load()?;
+    pub fn new(poll: osaka::Poll, shadow: identity::Address, config: config::Config) -> Result<Self, Error> {
```
[the remark this answers is not in the document]
Comment (ewilken) [reply src/conduit.rs]: Like
```
pub struct Config {
    pub secret:    identity::Secret,
    pub keepalive: Option<u16>,
    pub publish:   Option<PublisherConfig>,
    pub names:     HashMap<String, identity::Identity>,
    pub shadow:    Option<identity::Address>,
}
```
and
```
pub struct ConfigToml {
    secret:    Option<String>,
    keepalive: Option<u16>,
    publish:   Option<PublisherConfigToml>,
    authorize: Option<Vec<AuthorizationToml>>,
    names:     Option<HashMap<String, String>>,
    shadow:    Option<String>,
}
```
or some sort of separate `ConduitConfig` struct? I'm not sure if I fully understand that whole constellation of configs yet.

[the remark this answers is not in the document]
Comment (aep) [reply src/conduit.rs]: No. Like 

[[subscribe]]
shadow=

I.e. same as publish

[the remark this answers is not in the document]
Comment (ewilken) [reply src/conduit.rs]: I think I got it. Should a None `SubscriberConfig` in the config on conduit creation be a special error kind, a panic or just create a conduit that does nothing?

[the remark this answers is not in the document]
Comment (aep) [reply src/conduit.rs]: probably panic, since it makes no sense

**Pull request merged**